File: library/src/main/java/com/trinity/recording/service/impl/AudioTrackPlayerServiceImpl.java
package com.trinity.recording.service.impl;

import android.media.AudioManager;
import com.trinity.player.AudioTrackPlayer;
import com.trinity.recording.service.AbstractPlayerServiceImpl;
import com.trinity.recording.service.PlayerService;

public class AudioTrackPlayerServiceImpl extends AbstractPlayerServiceImpl
    implements PlayerService, PlayerService.OnCompletionListener {

  public AudioTrackPlayer mediaPlayer;

  public AudioTrackPlayerServiceImpl() {
    try {
      if (mediaPlayer == null) {
        mediaPlayer = new AudioTrackPlayer();
        mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
        mediaPlayer.setOnCompletionListener(this);
      }
    } catch (Exception e) {
      e.printStackTrace();
    }

  }

  @Override
  public boolean setAudioDataSource(int vocalSampleRate) {
    boolean result = false;
    if (mediaPlayer != null) {
      result = mediaPlayer.prepare(vocalSampleRate);
    }
    return result;
  }

  @Override
  public int getAccompanySampleRate() {
    try {
      if (mediaPlayer != null) {
        return mediaPlayer.getAccompanySampleRate();
      }
    } catch (IllegalStateException e) {
      e.printStackTrace();
    }
    return -1;
  }

  @Override
  public void start() {
    super.start();
    try {
      if (mediaPlayer != null) {
        mediaPlayer.start();
      }
    } catch (IllegalStateException e) {
      e.printStackTrace();
    }
  }

  @Override
  public void stop() {
    super.stop();
    try {
      if (mediaPlayer != null) {
        mediaPlayer.stop();
      }
    } catch (IllegalStateException e) {
      e.printStackTrace();
    }
  }

  @Override
  public int getPlayerCurrentTimeMills() {
    return mediaPlayer.getCurrentTimeMills();
  }

  @Override
  public void onCompletion() {

  }

  @Override
  public void setVolume(float volume, float accompanyMax) {
    mediaPlayer.setVolume(volume, accompanyMax);
  }

  @Override
  public void startAccompany(String path) {
    mediaPlayer.startAccompany(path);
  }

  @Override
  public void pauseAccompany() {
    mediaPlayer.pauseAccompany();
  }

  @Override
  public void resumeAccompany() {
    mediaPlayer.resumeAccompany();
  }

  @Override
  public void stopAccompany() {
    mediaPlayer.stopAccompany();
  }

  @Override
  public int getPlayedAccompanyTimeMills() {
    return mediaPlayer.getPlayedAccompanyTimeMills();
  }

  @Override
  public boolean isPlayingAccompany() {
    boolean ret = false;
    if (null != mediaPlayer) {
      ret = mediaPlayer.isPlayingAccompany();
    }
    return ret;
  }
}

File: library/src/main/java/com/trinity/recording/service/impl/AudioTrackPlayerServiceImpl.kt
package com.trinity.recording.service.impl

import android.media.AudioManager
import com.trinity.player.AudioTrackPlayer
import com.trinity.recording.service.PlayerService

open class AudioTrackPlayerServiceImpl : PlayerService, PlayerService.OnCompletionListener {

    private var mediaPlayer: AudioTrackPlayer? = null
    private var mPlaying = false

    override val isPlaying
        get() = mPlaying

    override val accompanySampleRate: Int
        get() {
            try {
                if (mediaPlayer != null) {
                    return mediaPlayer?.accompanySampleRate ?: 0
                }
            } catch (e: IllegalStateException) {
                e.printStackTrace()
            }

            return -1
        }

    override val playerCurrentTimeMills: Int
        get() = mediaPlayer?.currentTimeMills ?: 0

    override val playedAccompanyTimeMills: Int
        get() = mediaPlayer?.playedAccompanyTimeMills ?: 0

    override val isPlayingAccompany: Boolean
        get() {
            var ret = false
            if (null != mediaPlayer) {
                ret = mediaPlayer?.isPlayingAccompany ?: false
            }
            return ret
        }

    init {
        try {
            if (mediaPlayer == null) {
                mediaPlayer = AudioTrackPlayer()
                mediaPlayer?.setAudioStreamType(AudioManager.STREAM_MUSIC)
                mediaPlayer?.setOnCompletionListener(this)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun setAudioDataSource(vocalSampleRate: Int): Boolean {
        return mediaPlayer?.prepare(vocalSampleRate) ?: false
    }

    override fun start() {
        try {
            mPlaying = true
            mediaPlayer?.start()
        } catch (e: IllegalStateException) {
            e.printStackTrace()
        }

    }

    override fun stop() {
        try {
            mPlaying = false
            mediaPlayer?.stop()
        } catch (e: IllegalStateException) {
            e.printStackTrace()
        }

    }

    override fun onCompletion() {

    }

    override fun setVolume(volume: Float, accompanyMax: Float) {
        mediaPlayer?.setVolume(volume, accompanyMax)
    }

    override fun startAccompany(path: String) {
        mediaPlayer?.startAccompany(path)
    }

    override fun pauseAccompany() {
        mediaPlayer?.pauseAccompany()
    }

    override fun resumeAccompany() {
        mediaPlayer?.resumeAccompany()
    }

    override fun stopAccompany() {
        mediaPlayer?.stopAccompany()
    }
}
